server/src/main/java/org/elasticsearch/rest/RestRequest.java

```diff
@@ -173,6 +175,64 @@ public static RestRequest requestWithoutParameters(NamedXContentRegistry xConten
             requestIdGenerator.incrementAndGet());
     }
 
+    /**
+     * An http request can be accompanied with a compatible version indicating with what version a client is using.
+     * Only a major Versions are supported. Internally we use Versions objects, but only use Version(major,0,0)
+     * @return a version with what a client is compatible with.
+     */
+    public Version getCompatibleApiVersion(Version minimumRestCompatibilityVersion) {
+        if (/*headersValidation &&*/ isRequestingCompatibility()) {
+            return minimumRestCompatibilityVersion;
+        } else {
+            return Version.CURRENT;
+        }
+    }
+
+
+    private boolean isRequestingCompatibility() {
+      /*  String acceptHeader = header(CompatibleConstants.COMPATIBLE_ACCEPT_HEADER);
```

Comment:
If we pass around a function we can put this logic in the plugin. Calling the function or not can be determined if the header exists at all or not.

Reply:
see https://github.com/elastic/elasticsearch/pull/60220#discussion_r461545470